### ISSUE TITLE
High Memory Leak

### ISSUE THREAD

**Issue by cenobyte321** [winkerVSbecks/angular-pdf-viewer#55]
Hello,

I've been using this plugin for an Ionic Framework (based on AngularJS) app and it is leaking memory. 

Here's a very good analysis that proves the leak:
https://github.com/driftyco/ionic/issues/1096#issuecomment-206020609

The sample project the previous analysis references is this one:
https://github.com/cenobyte321/ionic-pdf-video-memory

**Comment by winkerVSbecks** [comment]
@cenobyte321 good point. I need to add a `pdfDoc.destroy()` on destroy. Or at least expose a destroy method. 

I noticed there's a bunch of stuff happening in the app you created: DOM manipulation and adding the video tags, etc. Could you run another test where you remove all that and just leave the PDF in the view. Just curious if that makes any impact.

**Comment by cenobyte321** [comment]
Hello @winkerVSbecks I've made another branch without the videos and with less operations:

https://github.com/cenobyte321/ionic-pdf-video-memory/tree/test-2-no-videos

The memory usage increments by transitioning between the two views but in a lower rate now. 

I still believe having the destroy() and clean() methods of PDF.js at least exposed in the wrapper could help to actually clean anything that is left by that plugin, and would be a very good upgrade.

One more thing, I am using in the project this pull request to render all pages from the PDF 

https://github.com/winkerVSbecks/angular-pdf-viewer/pull/51

Would adding the destroy and clean methods also solve this situation with that pull request?

**Comment by winkerVSbecks** [comment]
Yes, I'm absolutely going to add the destroy and clean methods. Was just curious if PDFJS did some of that automatically. Doesn't seem like it though.

**Comment by winkerVSbecks** [comment]
I'm sure the render all pages is increasing the memory leak. 1 vs multiple canvases. Since PDFJS methods still reference them the canvases are probably not being destroyed.

**Comment by mbavio** [comment]
@cenobyte321 If you are curious, I added this functionality in my fork and created a PR on this repo.

**Comment by mbavio** [comment]
Thanks for reporting btw, I was not aware of this issue!
